api/api/utils/moderation_lock.py

```diff
@@ -0,0 +1,123 @@
+import functools
+import time
+
+import django_redis
+import structlog
+from redis.exceptions import ConnectionError
+
+from api.models.moderation import get_moderators
+
+
+LOCK_PREFIX = "moderation_lock"
+TTL = 10  # seconds
+
+logger = structlog.get_logger(__name__)
+
+
+def handle_redis_exception(func):
+    @functools.wraps(func)
+    def wrapper(*args, **kwargs):
+        try:
+            return func(*args, **kwargs)
+        except ConnectionError:
+            return None
+
+    return wrapper
+
+
+class LockManager:
+    def __init__(self, media_type):
+        self.media_type = media_type
+
+    @handle_redis_exception
+    def prune(self) -> dict[str, set[str]]:
+        """
+        Delete all expired locks and get a mapping of usernames to
+        media items that have active locks.
```

Comment:
Just wanted to add credit for the idea of the ranked set for expiration :smile: 

```suggestion
        media items that have active locks.
        
        Kudos to this Google Group discussion for the solution using a ranked set
        https://web.archive.org/web/20211205091916/https://groups.google.com/g/redis-db/c/rXXMCLNkNSs
```

Reply:
Sure, added in 899de0a.